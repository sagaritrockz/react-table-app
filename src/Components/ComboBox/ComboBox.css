.combo-box {
    border: 1px solid #dee2e6;
    padding: 15px;
    display: inline-block;
    margin: 0 0 10px 0;
}
.combo-box-container {
    display: flex;
    justify-content: center;
}
.combo-box-container .select-custom {
    margin: 0 10px;
    width: 250px;
}