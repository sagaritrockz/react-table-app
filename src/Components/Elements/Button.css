.btn {
    display: inline-block;
    font-weight: 400;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    border: 1px solid transparent;
    padding: 5px 10px;
    font-size: 14px;
    line-height: 1.5;
    border-radius: 0.25rem;
    cursor: pointer;
    transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
        border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}
.btn-custom {
    margin: 0 5px;
}