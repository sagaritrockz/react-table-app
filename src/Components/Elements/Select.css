.select {
    display: inline-block;
    padding: 5px 10px;
    font-size: 14px;
    line-height: 1.5;
    color: #495057;
    background-color: #fff;
    background-clip: padding-box;
    border: 1px solid #ced4da;
    border-radius: 0.25rem;
    cursor: pointer;
    transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
    text-transform: capitalize;
}
.select-custom {
    padding: 5px 10px 8px;
    margin: 0 5px;
}
