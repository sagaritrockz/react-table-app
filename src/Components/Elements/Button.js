import React from "react"
import { string, func } from 'prop-types'
import "./Button.css"

const Button = ({ label, click, name }) => {
    return (
        <button className="btn btn-custom" onClick={click} name={name}>
            {label}
        </button>
    );
};

Button.defaultProps = {
    label: "",
    click: f => f,
    name: ""
};

Button.propTypes = {
    label: string.isRequired,
    click: func.isRequired,
    name: string.isRequired
};

export default Button;
