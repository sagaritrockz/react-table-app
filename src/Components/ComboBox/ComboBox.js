import React from "react"
import { array, string, func, object } from "prop-types"
import Select from "../Elements/Select"
import Button from "../Elements/Button"
import "./ComboBox.css"

const ComboBox = ({
    filters,
    apply,
    clearFilter,
    secondaryFilterVal,
    primaryFilterVal,
    updateState,
    secondaryFilterKey,
    primaryFilterKey,
    primaryFilters,
    secondaryFilters,
    secondaryFiltersKey
}) => {
    const onApply = () => {
        apply();
    };
    const onClear = () => {
        clearFilter();
    };
    const onPrimaryFilterChange = (key, value) => {
        updateState(key, value);
        populateSecondaryFilter(value);
    };
    const populateSecondaryFilter = value => {
        if (value < 1) {
            updateState(secondaryFiltersKey, []);
            updateState(secondaryFilterKey, 0);
        } else {
            updateState(
                secondaryFiltersKey,
                filters[primaryFilters[value - 1]]
            );
            updateState(
                secondaryFilterKey,
                filters[primaryFilters[value - 1]][0].id
            );
        }
    };

    return (
        <div className="combo-box-container">
            <div className="combo-box">
                <Select
                    list={primaryFilters}
                    value={primaryFilterVal}
                    change={onPrimaryFilterChange}
                    name={primaryFilterKey}
                    defaultOption="Select"
                />
                <Select
                    list={secondaryFilters}
                    value={secondaryFilterVal}
                    change={updateState}
                    name={secondaryFilterKey}
                    defaultOption="Select"
                />
                <Button click={onApply} label="Apply" name="apply" />
                <Button click={onClear} label="Clear" name="clear" />
            </div>
        </div>
    );
};

ComboBox.defaultProps = {
    filters: [],
    apply: f => f,
    clearFilter: f => f,
    secondaryFilterVal: "",
    primaryFilterVal: "0",
    updateState: f => f,
    secondaryFilterKey: "",
    primaryFilterKey: "",
    primaryFilters: [],
    secondaryFilters: [],
    secondaryFiltersKey: ""
};

ComboBox.propTypes = {
    filters: object.isRequired,
    apply: func.isRequired,
    clearFilter: func.isRequired,
    secondaryFilterVal: string.isRequired,
    primaryFilterVal: string.isRequired,
    updateState: func.isRequired,
    secondaryFilterKey: string.isRequired,
    primaryFilterKey: string.isRequired,
    primaryFilters: array.isRequired,
    secondaryFilters: array.isRequired,
    secondaryFiltersKey: string.isRequired
};

export default ComboBox;
