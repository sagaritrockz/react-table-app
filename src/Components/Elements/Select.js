import React from "react"
import "./Select.css"

const Select = ({ list, change, value, name, show }) => {

    const onChange = e => {
        e.preventDefault();
        change(e.target.name, e.target.value);
    };
    
    if (show) {
        return (
            <select
                className="select select-custom"
                onChange={onChange}
                name={name}
                value={value}
            >
                <option value={0}>Select</option>
                {list.map((item, index) => (
                    <option key={index} value={item.id ? item.id : index + 1} >
                        {item.value ? item.value : item}
                    </option>
                ))}
            </select>
        );
    } else {
        return (
            <select
                className="select select-custom"
                onChange={onChange}
                name={name}
                value={value}
            >
                {list.map((item, index) => (
                    <option key={index} value={item.id ? item.id : index + 1} >
                        {item.value ? item.value : item}
                    </option>
                ))}
            </select>
        );
    }
};

Select.defaultProps = {
    list: [],
    change: f => f, 
    value: '', 
    name: '',
    show: false
}

export default Select;
