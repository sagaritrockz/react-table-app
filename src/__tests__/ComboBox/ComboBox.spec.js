import React from 'react';
import Enzyme from 'enzyme';
import Adaptar from 'enzyme-adapter-react-16.3';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ComboBox from '../../Components/ComboBox/ComboBox';
import Select from "../../Components/Elements/Select";
import Button from "../../Components/Elements/Button";

Enzyme.configure({adapter: new Adaptar()});

describe('ComboBox', () => {
    it('should render correctly', () => {
        const 
            filters = {},
            onApply = function() {},
            clearFilter = function() {},
            searchValue = 0,
            searchOn = 0,
            updateState = function() {},
            searchValueKey = "_paginationDropdown",
            searchOnKey = "_page",
            primaryFilters = [],
            secondaryFilters = [],
            secondaryFiltersKey = "_secondaryFilters"

        const output = shallow(
            <ComboBox
                filters={filters}
                apply={onApply}
                clearFilter={clearFilter}
                secondaryFilterVal={String(searchValue)}
                primaryFilterVal={String(searchOn)}
                updateState={updateState}
                secondaryFilterKey={searchValueKey}
                primaryFilterKey={searchOnKey}
                primaryFilters={primaryFilters}
                secondaryFilters={secondaryFilters}
                secondaryFiltersKey={secondaryFiltersKey}
            />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});