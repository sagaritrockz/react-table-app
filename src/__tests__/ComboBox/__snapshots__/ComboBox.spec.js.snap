// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComboBox should render correctly 1`] = `
<div
  className="combo-box-container"
>
  <div
    className="combo-box"
  >
    <Select
      change={[Function]}
      defaultOption="Select"
      list={Array []}
      name="_page"
      value="0"
    />
    <Select
      change={[Function]}
      defaultOption="Select"
      list={Array []}
      name="_paginationDropdown"
      value="0"
    />
    <Button
      click={[Function]}
      label="Apply"
      name="apply"
    />
    <Button
      click={[Function]}
      label="Clear"
      name="clear"
    />
  </div>
</div>
`;
